pkg/controllers/resources/persistentvolumes/syncer.go

```diff
@@ -145,8 +145,15 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
 	if err != nil {
 		return ctrl.Result{}, err
 	} else if !sync {
-		ctx.Log.Infof("delete virtual persistent volume %s, because there is no virtual persistent volume claim with that volume", vPersistentVolume.Name)
-		return ctrl.Result{}, ctx.VirtualClient.Delete(ctx.Context, vObj)
+		if vPersistentVolume.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimDelete {
```

Comment:
@maheshbaliga Does it consider the "Default" reclaim policies for PV? Seems it has different defaults, depending on if the PV is created manually or dynamically. See: https://pkg.go.dev/k8s.io/api/core/v1#PersistentVolumeSpec